src/table/hooks/use-field-selection.ts

```diff
@@ -30,7 +40,18 @@ const useFieldSelection = (column: Column): UseFieldSelectionOutput => {
 
   useEffect(() => {
     if (!app || !app.getField || !column || !column.isDim) return;
-    app.getField(column.fieldId, layout.qStateName).then(setFieldInstance);
+    const { qLibraryId, fieldId } = column;
+    qLibraryId
+      ? app.createSessionObject(getListBoxSessionObject(qLibraryId, layout.qStateName)).then((listboxSessionObject) =>
+          setFieldInstance({
+            selectAll: () => listboxSessionObject.selectListObjectAll('/qListObjectDef'),
```

Comment:
Just a random thought, can we use model here instead of creating a session object or they are totally used for different purposes? [(This here from nebula is why i ask this question)](https://github.com/qlik-oss/nebula.js/blob/280f587b60698bd76919a0b72a249c7e747a1749/apis/nucleus/src/components/listbox/interactions/listbox-selection-toolbar.js#L22-L32)

Plus, we have direct access to model in line 35 through `useContextSelector(...)` call

Reply:
So you mean the model that we have access within table ([which comes form using `useModel` hook](https://github.com/qlik-oss/sn-table/blob/bbedb322dd622ba725d72d688c4866e964a26d40/src/index.ts#L43C14-L43C14)) and the model that nebula uses (useSessionModel -> [which ends up in here](https://github.com/qlik-oss/nebula.js/blob/280f587b60698bd76919a0b72a249c7e747a1749/apis/nucleus/src/hooks/useSessionModel.js#L23)) are different? 

The confusing part for me is they both are model, but how they (may) act differently in this case that would make us to create a new session object from scratch (for master dims -> which is distinguishable from a normal dim by `qLibraryId`)?

Reply:
The model that comes from the useModel hook is the model for the table. It is completely different from the model that is returned for a single field (or in this case the on the fly session object for a listbox)